feature/infra-pulumi

Zenemis opened the pull request:
Deployment tested with "pulumi up", **resources are online** (and free of cost :+1:) :  
- Resource group (_rg-nebulift-dev-westeurope-be503450_)
- App service (_wapp-nebulift-dev-westeurope-5b5d0dca_)
- Static web app (_stapp-nebulift-dev-westeurope-f3ecd0b8_)

Static Web App will require future setup of **repo**, **sitePath** and **apiPath** 
App Service will require future setup of **endpoint**

I created a Pulumi organization and a Pulumi stack (dev)
**I need to transfer its ownership to @TechWatching @theogom @fboitel**

Zenemis commented:
To test from your side : 
cd infra
dotnet restore
pulumi up (will probably not work without a Pulumi stack and a Pulumi account)

TechWatching commented on infra/MyStack.cs:
```diff
@@ -0,0 +1,81 @@
+                Name = "Free"
+            }
+        });
+
```
You probably will need to link the API to the static web app to make easy communication between them: https://www.pulumi.com/registry/packages/azure-native/api-docs/web/staticsitelinkedbackend/

Zenemis commented on infra/nebulift.csproj:
```diff
@@ -0,0 +1,14 @@
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net6.0</TargetFramework>
```
Done !

Zenemis commented:
Every change from previous request change applied : 
- Switched to .NET 8.0
- Nebulift.csproj -> Nebulift.infra.csproj
- Tiers pulled from config variables
- No longer automatic deploy of frontend on stapp
- Added support of backend-link (:-1: **paid plan is mandatory for this feature**)

theogom commented on infra/MyStack.cs:
```diff
@@ -0,0 +1,87 @@
+        var staticWebAppSkuName = config.Require("StaticWebAppSkuName");
+
+        // Create an Azure Resource Group
+        var resourceGroup = new ResourceGroup("rg-nebulift-dev-westeurope-");
```
We should parameterize the ressource names and add it in the configuration to be able to deploy different environments like development and production (even if we will not need it here)

theogom commented on infra/nebulift.infra.csproj:
```diff
@@ -0,0 +1,14 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
The csproj file should be named in Pascal case `Nebulift.Infra` to follow conventions

TechWatching commented on infra/MyStack.cs:
```diff
@@ -0,0 +1,101 @@
+
+class MyStack : Stack
+{
+    private string CreateResourceName(params string[] args){
```
No need for this method

TechWatching commented on infra/MyStack.cs:
```diff
@@ -0,0 +1,101 @@
+        var staticWebAppSkuName = config.Require("StaticWebAppSkuName");
+
+        // Create an Azure Resource Group
+        var resourceGroup = new ResourceGroup(CreateResourceName("rg", projectName, stackName));
```
just use string interpolation instead of an external method : 

```csharp
var resourceGroup = new ResourceGroup($"rg-{projectName}-{stackName}");
```

TechWatching commented on infra/MyStack.cs:
```diff
@@ -0,0 +1,101 @@
+        var appServicePlan = new AppServicePlan(CreateResourceName("wapp-plan", projectName, stackName), new AppServicePlanArgs
+        {
+            ResourceGroupName = resourceGroup.Name,
+            Location = resourceGroup.Location,
```
location will be automatically set in your resources because you have it in configuration. no need to explicitly set it.

TechWatching replied:
Same for the other resources

TechWatching commented on infra/MyStack.cs:
```diff
@@ -0,0 +1,101 @@
+        // Link front-end to back-end
+        // Only possible on paid plan
+        // TODO : replace if condition with better practice
+        if (staticWebAppSkuName == "Standard" || staticWebAppSkuName == "Dedicated") {
```
Just use standard, pricing is not high. Check with @fboitel and @theogom but it should be okay

TechWatching reviewed (approved):
Only a few changes to do and it should be okay.
